File: source/web/web.sph.commercial-space/App/services/Deposit.js
/// <reference path="domain.g.js" />
/// <reference path="../../Scripts/underscore.js" />
/// <reference path="../../Scripts/knockout-2.2.1.debug.js" />

bespoke.sphcommercialspace.domain.DepositPartial = function (model) {
    var calcPayment = function () {
        var sum = _(model.DepositPaymentCollection()).reduce(function (memo, val) {
            return memo + parseFloat(val.Amount());
        }, 0);

        return sum.toFixed(2);
    },
        calcBalance = function () {
            var paid = _(model.DepositPaymentCollection()).reduce(function (memo, val) {
                return memo + parseFloat(val.Amount());
            }, 0);
             var sum = model.Amount() - paid;
             return sum.toFixed(2);
        };
    
    return {
        DepositPaid: ko.computed(calcPayment),
        DepositBalance: ko.computed(calcBalance)
    };
};

File: source/domain/domain.commercialspace/Business/Deposit.cs
using System.Linq;
using System.Xml.Serialization;

namespace Bespoke.SphCommercialSpaces.Domain
{
   public partial class Deposit : Entity
    {
        [XmlIgnore]
        public decimal DepositPaid
        {
            get
            {
                var sum = this.DepositPaymentCollection.Sum(d => d.Amount);
                return sum;
            }
        }

        [XmlIgnore]
        public decimal DepositBalance
        {
            get
            {
                var balance = this.Amount - this.DepositPaid;
                return balance;
            }
        }
    }
}
